### DISCUSSION TITLE
Nixvim equivalent for `vim.b`

### DISCUSSION THREAD

**Discussion by jolars** [nix-community/nixvim#2579] [Q&A]
Maybe I am missing something, but is there no equivalent option for setting custom buffer-local variables via `vim.b`?

I guess I can just do (using the home manager module):

```nix
  programs.nixvim.files."after/ftplugin/tex.lua".extraConfigLua = ''
    vim.b.my_option = false
  '';
```

But if would be nice if there was something like the `globals` setting, so that I could do something like this instead:

```nix
  programs.nixvim.files."after/ftplugin/tex.lua".bufferVariables = {
    my_option = false;
  };
```

**Comment by traxys** [comment]
I don't think we currently have a wrapper, as no one had an use case in mind for a declarative config. If you have an exemple of how it could be used I think it could easily be implemented

**Comment by jolars** [comment]
The example and use case I had in mind is exactly the one above, setting buffer-local custom variables for certain filetypes. Many plugins use these kinds of variables and I use my own variables as well to specify selective behavior for certain filetypes. I realize that the gain is not huge, but it also seems easy to do by just using the globals implementation as a blueprint.
